internal/ignition/ignition.go

```diff
@@ -839,18 +843,26 @@ func uploadToS3(ctx context.Context, workDir string, cluster *common.Cluster, s3
 	return nil
 }
 
-func parseIgnitionFile(path string) (*config_31_types.Config, error) {
-	configBytes, err := ioutil.ReadFile(path)
+func ParseTo32(content []byte) (*config_32_types.Config, error) {
+	configv32, _, err := config_32.Parse(content)
 	if err != nil {
-		return nil, errors.Errorf("error reading file %s: %v", path, err)
+		configv31, _, err := config_31.Parse(content)
+		if err != nil {
+			return nil, errors.Errorf("error parsing ignition: %v", err)
+		}
+		configv32 = config_32_trans.Translate(configv31)
+		configv32.Ignition.Version = "3.1.0"
```

Comment:
What is the purpose of this line?

Reply:
This is what allows us to write out 3.1 json even though we're using a 3.2 type.
It's pretty much the heart of the strategy as the [translator changes the version to 3.2](https://github.com/coreos/ignition/blob/8ef8dc0c579202ea1228a7a264315f47599374e7/config/v3_2/translate/translate.go#L26)

Reply:
What we're saying here is that if what we got from the installer binary were 3.1 ignitions then we should write out 3.1 ignitions after editing them.

Reply:
I suppose this could also be [`config_31.MaxVersion.String()`](https://github.com/coreos/ignition/blob/8ef8dc0c579202ea1228a7a264315f47599374e7/config/v3_1/types/config.go#L22), but I think `3.1.0` will always be valid.

Reply:
This is the part I'm not getting - why do we want a 3.1.0 JSON? What entity still consumes / expects a 3.1 json?

Reply:
Also, why not just update that entity to expect 3.2 instead?

Reply:
> why do we want a 3.1.0 JSON? What entity still consumes / expects a 3.1 json?

OpenShift 4.6

Reply:
what will happen in cases ignition had content that exist only in 3.2 and cannot be translated back to 3.1?
wont this fail?
so you are not actually supporting 3.2 content?

Reply:
> This strategy leaves us open to potential bugs if we somehow start
using features from later versions of ignition either in our source
or from user customizations, but this solution is the best we can do
in the short term to support installing OpenShift 4.6, 4.7, and 4.8
in the same codebase.

Reply:
@yuvalk can u give 3.2 example? Why did u do the change and where will it break us

Reply:
> From PR description:
> 
> > This strategy leaves us open to potential bugs if we somehow start
> > using features from later versions of ignition either in our source
> > or from user customizations, but this solution is the best we can do
> > in the short term to support installing OpenShift 4.6, 4.7, and 4.8
> > in the same codebase.

Then we need a path forward to fixing this.
otherwise we wont be able to do things like disk encryption with AI, in both 4.7 and 4.8.

Reply:
Right now, this is only a concern with the ignition customization feature.
If the user provides us with ignition  3.2 and wants to install OCP 4.6, for example
We'll add validations for this later

Reply:
> @yuvalk can u give 3.2 example? Why did u do the change and where will it break us

yes, see the BZ attached to that PR
basically I was trying to enable disk encryption as described here
https://github.com/openshift/openshift-docs/blob/enterprise-4.7/modules/installation-special-config-encrypt-disk-tang.adoc
rhcos-4.7 already support only the new v3.2 luks config

Reply:
> Right now, this is only a concern with the ignition customization feature.
> If the user provides us with ignition 3.2 and wants to install OCP 4.6, for example
> We'll add validations for this later

the question is what will happen if user provide 3.2 ign and wants to install 4.7.
seems to me, you'll still try to convert that to 3.1 which, in my above example, will fail

Reply:
> the question is what will happen if user provide 3.2 ign and wants to install 4.7.
seems to me, you'll still try to convert that to 3.1 which, in my above example, will fail

This is not how this code works.
We maintain whatever version of ignition we read from the files generated by the installer.
So if the installer generates 3.2 we never change that to 3.1